runtime/sema/type.go

```diff
@@ -5427,226 +5505,18 @@ func checkSubTypeWithoutEquality(subType Type, superType Type) bool {
 		)
 
 	case *ReferenceType:
-		// References types are only subtypes of reference types
-
 		typedSubType, ok := subType.(*ReferenceType)
 		if !ok {
 			return false
 		}
 
-		// An authorized reference type `auth &T`
-		// is a subtype of a reference type `&U` (authorized or non-authorized),
-		// if `T` is a subtype of `U`
-
-		if typedSubType.Authorized {
-			return IsSubType(typedSubType.Type, typedSuperType.Type)
-		}
-
-		// An unauthorized reference type is not a subtype of an authorized reference type.
-		// Not even dynamically.
-		//
-		// The holder of the reference may not gain more permissions.
-
-		if typedSuperType.Authorized {
+		// the authorization of the subtype reference must be usable in all situations where the supertype reference is usable
+		if !typedSuperType.Authorization.PermitsAccess(typedSubType.Authorization) {
 			return false
 		}
 
-		switch typedInnerSuperType := typedSuperType.Type.(type) {
```

Comment:
!

Reply:
Could you please delete the file? It won't be accurate anymore after the merge of this PR